libs/core/langchain_core/schema/chat_model.py

```diff
@@ -0,0 +1,48 @@
+from abc import ABC, abstractmethod
+from typing import Any, List, Optional
+
+from langchain_core.callbacks import (
+    AsyncCallbackManagerForLLMRun,
+    CallbackManagerForLLMRun,
+)
+from langchain_core.messages import BaseMessage
+from langchain_core.outputs import ChatResult
+
+
+# TODO: this is a work in progress
+# TODO: do we need other methods from BaseChatModel or BaseLanguageModel?
+class ChatModelInterface(ABC):
+    """Interface for Chat Model.
+
+    The default implementation of this interface is :class:`BaseChatModel`.
+    """
+
+    @abstractmethod
+    def _call(
```

Comment:
All chat models should be runnables, so we'd like to enforce that by design

Reply:
Yes. But runnables have separate interfaces and Base classes derive the base interface + runnable interface.